src/Time.ts

```diff
@@ -1,8 +1,24 @@
 import { Effect } from './Effect'
-import { handle, ok } from './Fx'
+import { Fx, handle, ok } from './Fx'
+import { ScheduledTask, TimeStep, TimeoutDisposable } from './internal/time'
 
-export class Now extends Effect('fx/Time')<void, number> { }
+export class Now extends Effect('fx/Time/Now')<void, number> { }
```

Comment:
I would love to use better result types for Now and Monotonic.  Using `number` for both is confusing and not very safe.  It'd be great to use Temporal types, like Instant for Now and Duration for Monotonic, but it's [still at stage 3](https://github.com/tc39/proposal-temporal).

Any ideas?

I guess we could roll our own Temporal-compatible types.

Reply:
I agree, it'd be nice to use Temporal. Maybe on the easy end, we could just hand-roll some branded types for them. I could also imagine just a simple types like

```ts
interface Duration { milliseconds: number  }

interface UnixTime { millisecondsSinceEpoch: number }
```

but those won't work at all with and mathematical operators by default and increases memory usage.

Reply:
Branded types might be ok, at least for communicating intent.  I'm wary of forcing people to use a library-specific type, which is another reason Temporal is so appealing.  Your interface idea might work if they mimic Temporal, maybe like:

```ts
interface Instant { readonly epochMilliseconds: number }
interface Duration { readonly milliseconds: number }
```

They couldn't be passed directly to Temporal functions that require an Instant or Duration, but the Duration type would work in place of Temporal's DurationLike.

I could also just see going with number for now (even though it's a bummer in the short term)

Reply:
Not sure what I think of it, but looking a bit at the temporal docs, I notice that `new Instant(bigintOfNanoSeconds)`. Maybe they could be different just by using `number` for Duration, and `bigint` for Instant

Reply:
I wrestled with that idea, too.  I think the main downside is that it'll make Time.now values slightly less ergonomic to consume.  Although, that's a better situation than using our own Temporal-like objects.

I'll try it and we can see how it looks.

Reply:
Switched `Now` to `bigint` in https://github.com/briancavalier/fx/pull/14/commits/0545877095dccf2bfdfef22a955e6e28c3022a34.  I think it works out nicely, and at least heads in the direction of Temporal.